cpp/src/parquet/column_writer.cc

```diff
@@ -2050,6 +2050,7 @@ Status TypedColumnWriterImpl<FLBAType>::WriteArrowDense(
     WRITE_SERIALIZE_CASE(FIXED_SIZE_BINARY, FixedSizeBinaryType, FLBAType)
     WRITE_SERIALIZE_CASE(DECIMAL128, Decimal128Type, FLBAType)
     WRITE_SERIALIZE_CASE(DECIMAL256, Decimal256Type, FLBAType)
+    WRITE_SERIALIZE_CASE(HALF_FLOAT, FixedSizeBinaryType, FLBAType)
```

Comment:
This would resolve the segfault but doesn't really feel elegant.
```suggestion
    case ::arrow::Type::HALF_FLOAT: {
      auto array_data = array.data();
      const auto& arr = ::arrow::FixedSizeBinaryArray(
          ::arrow::fixed_size_binary(2), array.length(), array_data->buffers[1],
          array_data->buffers[0], array.null_count(), array.offset());
      return WriteArrowSerialize<FLBAType, ::arrow::FixedSizeBinaryType>(
          arr, num_levels, def_levels, rep_levels, ctx, this, maybe_parent_nulls);
    }
```